tests/bluetooth/bsim_bt/bsim_test_per_adv/src/per_adv_syncer.c

```diff
@@ -134,9 +157,43 @@ static void term_cb(struct bt_le_per_adv_sync *sync,
 	SET_FLAG(flag_per_adv_sync_lost);
 }
 
+static void recv_cb(struct bt_le_per_adv_sync *recv_sync,
+		    const struct bt_le_per_adv_sync_recv_info *info,
+		    struct net_buf_simple *buf)
+{
+	if (TEST_FLAG(flag_per_adv_recv)) {
+		return;
+	}
+
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	uint8_t buf_data_len;
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	printk("PER_ADV_SYNC[%u]: [DEVICE]: %s advertisment received\n",
+	       bt_le_per_adv_sync_get_index(recv_sync), le_addr);
+
+	buf_data_len = (uint8_t)net_buf_simple_pull_le16(buf);
+
+	if (buf->data[0]-1 != ad[0].data_len ||
+		memcmp(buf->data, ad[0].data, ad[0].data_len)) {
+		FAIL("Unexpected adv data received\n");
+	}
+
+	net_buf_simple_pull(buf, ad[0].data_len);
+	buf_data_len = (uint8_t)net_buf_simple_pull_le16(buf);
+
+	if (buf->data[0]-1 != ad[1].data_len ||
+		memcmp(buf->data, ad[1].data, ad[1].data_len)) {
+		FAIL("Unexpected adv data received\n");
+	}
```

Comment:
Suggest to use `bt_data_parse` as that will also validate the length checks, and make it easier to perform the comparison for each LTV structure